apis/monitoring/v1beta1/collector_config_types.go

```diff
@@ -0,0 +1,52 @@
+package v1beta1
+
+import (
+	"github.com/rancher/opni/plugins/metrics/pkg/apis/node"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PrometheusDiscovery struct {
+	NamespaceSelector      []string
+	ServiceMonitorSelector []string
+	PodMonitorSelector     []string
+	ProbeSelector          []string
+}
+
+type CollectorConfigSpec struct {
+	PrometheusDiscovery PrometheusDiscovery `json:"prometheusDiscovery,omitempty"`
+	RemoteWriteEndpoint string              `json:"remoteWriteEndpoint,omitempty"`
+	OtelSpec            node.OTELSpec       `json:"otelSpec,omitempty"`
```

Comment:
copy the OTELSpec struct (or an equivalent refactor), such that we don't have to import plugins/metrics from here